pkg/scheduler/api/resource_info.go

```diff
@@ -212,6 +212,29 @@ func (r *Resource) IsZero(rn v1.ResourceName) bool {
 	}
 }
 
+// AddDepends is used to add two given resources
+// e.g.1 r resource is <cpu 2000.00, memory 1000.00, nvidia.com/gpu 0>
+// rr resource is <cpu 3000.00, memory 2000.00 , nvidia.com/gpu 1>
+// return r resource is <cpu 5000.00, memory 3000.00, nvidia.com/gpu 0>
+// e.g.2 r resource is <cpu 2000.00, memory 1000.00>
+// rr resource is <cpu 3000.00, memory 2000.00 , nvidia.com/gpu 1>
+// return r resource is <cpu 5000.00, memory 3000.00, nvidia.com/gpu 0>
+func (r *Resource) AddDepends(rr *Resource) *Resource {
```

Comment:
Why not reuse function `Add`?

Reply:
The function logic is to not add ScalarResources when one of its properties is 0 or empty;